### PR TITLE
readme: update with 2.0 news!

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This serves as a reminder to do this before we tag 2.0.

### Testing Strategy

This pull request was tested by...

### TODO or Help Wanted

This pull request still needs...

### Documentation Updated

- [ ] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,22 +17,20 @@ protection units to isolate applications from each other and the kernel.
...
+more detail.
 
-The following combinations between kernel and userspace should work:
-- Tock on master, with libtock-c on master.
-- Tock on release v1.6, libtock-c on v1.6.
-- Tock on release v1.6, libtock-rs on master.
+- Revamped system call interface.
+- other
 
-Note, libtock-rs does not currently support Tock 2.0, but will before 2.0 is released.
```
Comment (hudson-ayers) [inline README.md:33]: We should probably keep a note that only libtock-c will work with tock 2.0 for the immediate future

Comment (ppannuto) [reply README.md:33]: I think we should have a tracking issue in the libtock-rs repo (or github project board? or equivalent) that this points to that lets people see what the status of libtock-rs for 2.0 is -- @jrvanwhy happy to defer to whatever you think would be best here, but something that gets updated as progress is made would be valuable I think

Comment (jrvanwhy) [reply README.md:33]: I created a tracking issue: https://github.com/tock/libtock-rs/issues/322. Its initial state is "everything I haven't done as of right now".

File: README.md [hunk 2]
```diff
@@ -17,22 +17,20 @@ protection units to isolate applications from each other and the kernel.
...
-
-Tock will continue to make this easier as we work towards the 2.0 release.
+As 2.0 includes many breaking changes, to use the new kernel you will need to
+ensure you have updated versions of userspace apps and the various Tock tools.
+We recommend if you are using git that you do a `git pull`, and if you are using
+tagged releases be sure to update to the 2.0 release.
```
Comment (bradjc) [inline README.md:33]:
```suggestion
tagged releases be sure to update to the 2.0 release.

Initially, only [libtock-c](https://github.com/tock/libtock-c) is compatible with
Tock 2.0. Work on supporting Tock 2.0 for libtock-rs is [under
development](https://github.com/tock/libtock-rs/issues/322).
```

File: README.md [hunk 3]
```diff
@@ -17,22 +17,24 @@ protection units to isolate applications from each other and the kernel.
...
+Tock 2.0!
+---------
 
-Note! Tock is currently switching from version 1.6 to 2.0 which includes
-breaking changes in the syscall interface. This means applications (for example
-from libtock-c) need to match the updated syscall interface. If the application
-does not match the kernel interface the app will likely fault very quickly.
+Tock is now on its second major release! Here are some 2.0 highlights, and see
+the [release notes](https://github.com/tock/tock/releases/tag/release-2.0) for
+more detail.
```
Comment (ppannuto) [inline README.md:25]:
```suggestion
more detail. Or the [changelog](CHANGELOG.md#new-in-20) for the complete set of changes.
```

File: README.md [hunk 4]
```diff
@@ -17,22 +17,24 @@ protection units to isolate applications from each other and the kernel.
...
 
-The following combinations between kernel and userspace should work:
-- Tock on master, with libtock-c on master.
-- Tock on release v1.6, libtock-c on v1.6.
-- Tock on release v1.6, libtock-rs on master.
+- Revamped system call interface.
+- other
```
Comment (ppannuto) [inline README.md:28]:
```suggestion
```

Comment (bradjc) [reply README.md:28]: Here are some highlights!

.
..
..

.


_crickets_

**Pull request merged**